ooasp/REST/solver_api.py

```diff
@@ -0,0 +1,913 @@
+            name, t1, t2 = assoc_data[-1].split(",")
+            if name not in specializations.keys():  # if there does not exist a specialisation it is the leaf
+                res.append(fact)
+    return res
+
+
+def parse_assumptions(a):
+    """
+    parses the model and only returns load-relevant facts.
+    """
+    res = []
+    for fact in a:
+        if "ooasp_isa" in fact:
+            res.append(f"{fact}.")
+        if "ooasp_attr_value" in fact:
+            res.append(f"{fact}.")
+        if "ooasp_associated" in fact:
+            assoc_data = fact.split("(")
+            name, t1, t2 = assoc_data[-1].split(",")
+            if name not in specializations.keys():  # if there does not exist a specialisation it is the leaf
+                res.append(f"{fact}.")
+    return res
+
+
+def initialise_solver(solver, data):
+    global selected_domain
+    selected_domain = os.path.split(data.domain)[0]
+
+    object_list = data.objects.split(",") if data.objects != "" else []
+    data.prio_associations = data.prio_associations.split(",")
+    solver.ctl.load(data.domain)
+    solver.load_base()
+    solver.smart_complete()
+    # check if a model with domain info was created
+    if solver.model is None:
+        return Response(message="Initial domain solve failed.", data=None).build()
+
+    solver.initial_objects = object_list
+    solver.associations_with_priority = data.prio_associations
+
+    solver.create_initial_objects()
+
+    global setup_flag
+    setup_flag = True
+
+    load_known_associations()
+    load_known_names()
+    load_known_attributes()
+    load_specializations()
+    # functions above can be consolidated into one in future to improve efficiency
+
+    return Response(message="Solver was initialised.", data=solver.__dict__).build()
+
+
+def _new_attr(node, attr):
+    global solver
+    found_val = None
+    for fact in solver.assumptions:
+        if "ooasp_attr_value" in fact:
+            val_info = fact.split("(")[-1].replace(")", "")
+            attr_name, target, value = val_info.split(",")
+            print(fact)
+            if (str(node["id"]) == str(target)) and (attr["name"] == attr_name):
+                found_val = value
+                break
+    vals = set()
+    vals.add(attr["value"])
+    attr_dict = {
+        "name": attr["name"],
+        "values": vals,
+        "active_value": found_val,
+        "object_id": node["id"]
+    }
+    for exisiting_attr in node["data"]["attributes"]:
+        if exisiting_attr["name"] == attr["name"] and exisiting_attr["object_id"] == node["id"]:
+            return
+
+    node["data"]["attributes"].append(attr_dict)
+    return
+
+
+def export_as_file(path):
+    global solver
+    with open(path, "w+") as f:
+        for assumption in solver.assumptions:
+            f.write(f"{assumption}.\n")
+
+
+def represent_as_graph():
+    """
+    Represents list of objects and associations a collection of nodes and edges.
```

Comment:
Not just nodes and edges, there seem to be also colours and positions and stuff... Please expand the docstring a little bit.